Grammar change to aid readability. Also added hyperlink to styled-com… (#20342)
gatsbyjs/gatsby: docs/docs/what-you-dont-need-plugins-for.md

- Importing JavaScript packages that provide general functionality, such as `lodash` or `axios`
- Integrating visualization libraries, such as `Highcharts` or `d3`.

> As a general rule, you may use any npm package you might use while working on a JavaScript or React application ,when deciding to work with Gatsby. What plugins offer is a prepackaged integration into the core Gatsby APIs to save you time and energy, with minimal configuration.

Reviewer: ```suggestion
As a general rule, any npm package you might use while working on another JavaScript or React application can also be used with a Gatsby application. What plugins offer is a prepackaged integration into the core Gatsby APIs to save you time and energy, with minimal configuration.
```

Author: I removed the yarn.lock from the commit

> A good use case would be using `Styled Components`, you could manually render the `Provider` component near the root of your application, or you could just use [gatsby-plugin-styled-components ](https://www.gatsbyjs.org/packages/gatsby-plugin-styled-components/)which takes care of this step for you in addition to any other difficulties you may run into configuring Styled Components to work with server side rendering.

Reviewer: ```suggestion
A good use case would be using `Styled Components`, you could manually render the `Provider` component near the root of your application, or you could use [gatsby-plugin-styled-components](/packages/gatsby-plugin-styled-components/) which takes care of this step for you in addition to any other difficulties you may run into when configuring Styled Components to work with server-side rendering.
```
